Latest version redirection

tenkoma opened the pull request:
This PR is a proposal to prepare an alias URL to the latest version in the API document site. Please review.

## Problem to solve

Currently, when linking to an API document from a website or markdown Docs, you need to specify a URL containing a minor version.
Even if there is a version upgrade (eg 3.0 -> 3.1) the link will not change.

* eg. https://book.cakephp.org/3.0/en/views/helpers/url.html (At the bottom, the link "Router::url" is transferred to the API document of version 3.0) 
* eg. https://github.com/cakephp/datasource/blob/master/README.md (At the bottom, the link "official API documentation" is transferred to the API document of version 3.2) 

I thought that if you had an alias that always redirects to the latest API document when upgrading, documents will be easier to maintain.
At the same time I set the transfer link to https.

If redirecting automatically to the latest version is inconvenient, please dismiss this suggestion.

## How it works

When the latest version is 3.4 and 2.9

* https://api.cakephp.org/ -> https://api.cakephp.org/3.4/ (Same as current)
* https://api.cakephp.org/3.latest -> https://api.cakephp.org/3.4/
* https://api.cakephp.org/3.latest/ -> https://api.cakephp.org/3.4/ (It works even if there is a slash at the end of the URL)
* https://api.cakephp.org/2.latest -> https://api.cakephp.org/2.9/
* https://api.cakephp.org/3.latest/class-Cake.ORM.Table.html -> https://api.cakephp.org/3.4/class-Cake.ORM.Table.html (You can also redirect the URL of the sub directory content)

josegonzalez commented on nginx.conf:
```diff
@@ -5,12 +5,23 @@ server {
+        rewrite ^/$ https://api.cakephp.org/${cake3_latest}/ permanent;
+    }
+
+    location ~ ^/3\.latest(/?.*)$ {
```
@markstory duplicate content?

markstory replied:
No as shorter versions of what is proposed here.

tenkoma commented:
@markstory Thanks to Feedback, I added follow rules.

* `/3.x/` -> `/3.4/`
* `/2.x/` -> `/2.9/`

And, I changed the redirect HTTP code to 303 (See Other). Because 301 (Moved Permanently) is cached by the browser, there is a possibility of redirecting to the old site when upgrading.

> I wonder if this could be `3.x` ,`2.x` and **`latest`** pointing to the most recent release?

Does this mean you also want a rule `/latest/` -> `/3.4/` ?

markstory commented:
Great work @tenkoma. I thought more about the `/latest` alias and I'm not sure it is a great idea. These new aliases will let us have more stable links in the book which is :+1:
